internal/cluster/cluster.go

```diff
@@ -158,3 +161,24 @@ func marshalBase64(in interface{}) ([]byte, error) {
 
 	return buffer.Bytes(), nil
 }
+
+// AgentServiceAccount secret takes a service account json and formats it as a
+// k8s secret.
+func AgentServiceAccountSecret(keyData []byte) *corev1.Secret {
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "agent-credentials",
+			Namespace: "jetstack-secure",
```

Comment:
I think the namespace will probably need to be modifyable. However, perhaps that can be done in a different PR when/if that gets asked for